federation/cmd/federation-apiserver/app/federation.go

```diff
@@ -22,20 +22,33 @@ import (
 	"k8s.io/apiserver/pkg/registry/generic"
 	"k8s.io/apiserver/pkg/registry/rest"
 	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/server/storage"
 	"k8s.io/kubernetes/federation/apis/federation"
-	"k8s.io/kubernetes/pkg/api"
-
 	_ "k8s.io/kubernetes/federation/apis/federation/install"
+	fedv1beta1 "k8s.io/kubernetes/federation/apis/federation/v1beta1"
 	clusteretcd "k8s.io/kubernetes/federation/registry/cluster/etcd"
+	"k8s.io/kubernetes/pkg/api"
 )
 
-func installFederationAPIs(g *genericapiserver.GenericAPIServer, optsGetter generic.RESTOptionsGetter) {
-	clusterStorage, clusterStatusStorage := clusteretcd.NewREST(optsGetter)
-	federationResources := map[string]rest.Storage{
-		"clusters":        clusterStorage,
-		"clusters/status": clusterStatusStorage,
+func installFederationAPIs(g *genericapiserver.GenericAPIServer, optsGetter generic.RESTOptionsGetter, apiResourceConfigSource storage.APIResourceConfigSource) {
+	groupName := federation.GroupName
+	if !apiResourceConfigSource.AnyResourcesForGroupEnabled(groupName) {
+		glog.V(1).Infof("Skipping disabled API group %q", groupName)
```

Comment:
This also looks like a lot of boilerplate code that should be done in a generic way somehow. The previously completely declarative style with one InstallAPIGroup style turned into some complex logic for each group. This will drift apart pretty fast.